pkg/s3/error.go

```diff
@@ -0,0 +1,9 @@
+package s3
+
+import "github.com/bacalhau-project/bacalhau/pkg/models"
+
+const S3_PUBLISHER_COMPONENT = "S3PUB"
+
+func NewErrBadS3Request(msg string) *models.BaseError {
```

Comment:
nit on naming here and elsewhere, would prefer:
`NewBadS3RequestError`

and more generally:
`New<Type>Error`